framework/src/media/InputHandler.cpp

```diff
@@ -195,10 +179,43 @@ void InputHandler::onBufferUpdated(ssize_t change, size_t current)
 	}
 }
 
+size_t InputHandler::getAvailSpace()
+{
+	if (mDemuxer) {
+		return mDemuxer->getAvailSpace();
+	}
+
+	if (mDecoder) {
+		return mDecoder->getAvailSpace();
+	}
+
+	// return PCM buffer space size
+	return mBufferWriter->sizeOfSpace();
+}
+
 ssize_t InputHandler::writeToStreamBuffer(unsigned char *buf, size_t size)
 {
 	assert(buf != nullptr);
 
+	if (mDemuxer) {
+		mDemuxer->pushData(buf, size);
+
+		if (!mDemuxer->isReady()) {
+			if (mDemuxer->prepare() < 0) {
+				medwdbg("Prepare demuxer failed!\n");
+				return size;
+			}
+		}
+
+		auto lenES  = mDemuxer->pullData(buf, size);
```

Comment:
pullData's return type is ssize_t, is there any reason for auto?

Reply:
Fixed, thanks